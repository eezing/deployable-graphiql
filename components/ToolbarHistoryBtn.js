import React from 'react';
import PropTypes from 'prop-types';
import GraphiQL from 'graphiql';

const ToolbarHistoryBtn = ({ graphiqlRef }) => {
  function _onClick() {
    graphiqlRef.current.handleToggleHistory();
  }

  return <GraphiQL.Button onClick={_onClick} label="History" title="History" />;
};

ToolbarHistoryBtn.propTypes = {
  graphiqlRef: PropTypes.object.isRequired
};

export default ToolbarHistoryBtn;
