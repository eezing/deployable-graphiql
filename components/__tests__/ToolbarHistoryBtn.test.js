import React from 'react';
import renderer from 'react-test-renderer';
import ToolbarHistoryBtn from '../ToolbarHistoryBtn';

test('ToolbarHistoryBtn snapshot', () => {
  const dummyRef = React.createRef();
  const component = renderer.create(
    <ToolbarHistoryBtn graphiqlRef={dummyRef} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
