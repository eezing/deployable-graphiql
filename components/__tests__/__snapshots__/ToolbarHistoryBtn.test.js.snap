// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarHistoryBtn snapshot 1`] = `
<a
  className="toolbar-button"
  onClick={[Function]}
  onMouseDown={[Function]}
  title="History"
>
  History
</a>
`;
